frontend/src/pages/LogsPage/SearchForm/SearchForm.tsx

```diff
@@ -59,8 +89,263 @@ const SearchForm = ({
 					/>
 				</Box>
 			</Box>
-		</form>
+		</Form>
 	)
 }
 
 export { SearchForm }
+
+const Search: React.FC<{
+	keys?: GetLogsKeysQuery['logs_keys']
+}> = ({ keys }) => {
+	const formState = useForm()
+	const { query } = formState.values
+	const { project_id } = useParams()
```

Comment:
I think it's mostly because we can handle massaging the data in the parent component. For example, we're setting that default value on `query`: https://github.com/highlight/highlight/blob/6dd92eda641a27854b4f2f89447b58a78f9f1852/frontend/src/pages/LogsPage/LogsPage.tsx#L48

For `project_id`, the type that we get back from `useParams` is `string | undefined`. It would be great if the parent component handles forcing this into a `string`. That way the child components don't have to do this weird `project_id!` syntax:

https://github.com/highlight/highlight/blob/6dd92eda641a27854b4f2f89447b58a78f9f1852/frontend/src/pages/LogsPage/SearchForm/SearchForm.tsx#L53-L57

Reply:
Yeah, that is annoying that we enforce `project_id` exists in our routing but it's still potentially `undefined` when we call `useParams`. I also just remembered we have the `useProjectId` hook. I think using that works around this problem.